x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pipelines/nlp/NerProcessor.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.ml.inference.pipelines.nlp;
+
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.xpack.core.ml.inference.deployment.PyTorchResult;
+import org.elasticsearch.xpack.ml.inference.pipelines.nlp.tokenizers.BertTokenizer;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class NerProcessor extends NlpPipeline.Processor {
+
+    public enum Entity {
+        NONE, MISC, PERSON, ORGANISATION, LOCATION
+    }
+
+    // Inside-Outside-Beginning (IOB) tag
+    private enum IobTag {
+        O(Entity.NONE),                 // Outside of a named entity
+        B_MISC(Entity.MISC),            // Beginning of a miscellaneous entity right after another miscellaneous entity
```

Comment:
I think the comments are backwards for B and I. I is inner so that's the tag that follows a B, "right after another ... entity"

Reply:
Wikipedia mentions and IOB2 format I think that is what you are referring to here

https://en.wikipedia.org/wiki/Inside%E2%80%93outside%E2%80%93beginning_(tagging)